core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -85,7 +85,7 @@
   private final LinkedList<UnsafeSorterSpillWriter> spillWriters = new LinkedList<>();
 
   // These variables are reset after spilling:
-  @Nullable private volatile UnsafeInMemorySorter inMemSorter;
+  @VisibleForTesting @Nullable volatile UnsafeInMemorySorter inMemSorter;
```

Comment:
This is slightly hair raising. Can you try to find a different - less tricky path - to test this?

Reply:
@hvanhovell , is it acceptable to a method indicating if the next inset is going to spill? Can make this @VisibleForTest and prefixed with something like testOnly